etl-jobs/src/main/scala/org/sunbird/analytics/jobs/UserCacheIndexer.scala

```diff
@@ -284,11 +289,38 @@ object UserCacheIndexer extends Serializable {
         .save()
     }
 
-    val res1 = time(populateUserData())
-    val res2 = time(denormUserData())
-    val totalTimeTaken = (res1._1 + res2._1).toDouble/1000
-    Console.println("Time taken for individual steps:", "stage1", res1._1, "stage2", res2._1)
-    Console.println("Time taken for complete script:", totalTimeTaken);
+    if (!populateAnonymousData.equalsIgnoreCase("true")) {
+      val res1 = time(populateUserData())
+      val res2 = time(denormUserData())
+      val totalTimeTaken = (res1._1 + res2._1).toDouble/1000
+      Console.println("Time taken for individual steps:", "stage1", res1._1, "stage2", res2._1)
+      Console.println("Time taken for complete script:", totalTimeTaken);
+    } else {
+      val sqlContext = new SQLContext(spark.sparkContext)
+      import sqlContext.implicits._
+
+      val anonymousDataRDD = spark.sparkContext.fromRedisKV("*").persist(StorageLevel.MEMORY_ONLY)
+      val filteredData = anonymousDataRDD.map{f => (f._1, JSONUtils.deserialize[Map[String, AnyRef]](f._2))}.filter(f => f._2.getOrElse("usersignintype", "").equals("Anonymous"))
```

Comment:
No need to persist the anonymous data frame

Reply:
Done